stdlib/_interpchannels.pyi

```diff
@@ -0,0 +1,86 @@
+from _typeshed import structseq
+from typing import Final, Literal, SupportsIndex, TypeVar, final
+from typing_extensions import Buffer, Self
+
+_T = TypeVar("_T")
+
+class ChannelError(RuntimeError): ...
+class ChannelClosedError(ChannelError): ...
+class ChannelEmptyError(ChannelError): ...
+class ChannelNotEmptyError(ChannelError): ...
+class ChannelNotFoundError(ChannelError): ...
+
+# Mark as final, since instantiating ChannelID is not supported.
+@final
+class ChannelID:
+    @property
+    def end(self) -> Literal["send", "recv", "both"]: ...
+    @property
+    def send(self) -> Self: ...
+    @property
+    def recv(self) -> Self: ...
+    def __eq__(self, other: object) -> bool: ...
+    def __ge__(self, other: ChannelID) -> bool: ...
+    def __gt__(self, other: ChannelID) -> bool: ...
+    def __hash__(self) -> int: ...
+    def __index__(self) -> int: ...
+    def __int__(self) -> int: ...
+    def __le__(self, other: ChannelID) -> bool: ...
+    def __lt__(self, other: ChannelID) -> bool: ...
+    def __ne__(self, other: object) -> bool: ...
+
+@final
+class ChannelInfo(structseq[int], tuple[bool, bool, bool, int, int, int, int, int]):
+    __match_args__: Final = (
+        "open",
+        "closing",
+        "closed",
+        "count",
+        "num_interp_send",
+        "num_interp_send_released",
+        "num_interp_recv",
+        "num_interp_recv_released",
+    )
+    @property
+    def open(self) -> bool: ...
+    @property
+    def closing(self) -> bool: ...
+    @property
+    def closed(self) -> bool: ...
+    @property
+    def count(self) -> int: ...  # type: ignore[override]
+    @property
+    def num_interp_send(self) -> int: ...
+    @property
+    def num_interp_send_released(self) -> int: ...
+    @property
+    def num_interp_recv(self) -> int: ...
+    @property
+    def num_interp_recv_released(self) -> int: ...
+    @property
+    def num_interp_both(self) -> int: ...
+    @property
+    def num_interp_both_recv_released(self) -> int: ...
+    @property
+    def num_interp_both_send_released(self) -> int: ...
+    @property
+    def num_interp_both_released(self) -> int: ...
+    @property
+    def recv_associated(self) -> bool: ...
+    @property
+    def recv_released(self) -> bool: ...
+    @property
+    def send_associated(self) -> bool: ...
+    @property
+    def send_released(self) -> bool: ...
+
+def create() -> ChannelID: ...
+def destroy(cid: SupportsIndex) -> None: ...
+def list_all() -> list[ChannelID]: ...
+def list_interpreters(cid: SupportsIndex, *, send: bool) -> list[int]: ...
+def send(cid: SupportsIndex, obj: object, *, blocking: bool = True, timeout: float | None = None) -> None: ...
+def send_buffer(cid: SupportsIndex, obj: Buffer, *, blocking: bool = True, timeout: float | None = None) -> None: ...
+def recv(cid: SupportsIndex, default: _T | None = None) -> object | _T: ...
```

Comment:
I'm not sure this is quite accurate... if you don't pass a value to the `default` parameter, it doesn't default to `None`, it just raises an exception:

```pycon
>>> import _interpchannels
>>> _interpchannels.create()
ChannelID(0)
>>> _interpchannels.recv(0)
Traceback (most recent call last):
  File "<python-input-2>", line 1, in <module>
    _interpchannels.recv(0)
    ~~~~~~~~~~~~~~~~~~~~^^^
_interpchannels.ChannelEmptyError: channel 0 is empty
```

But if you do provide a default, then the default value is returned:

```pycon
>>> _interpchannels.create()
ChannelID(1)
>>> _interpchannels.recv(1, 'foo')
'foo'
>>> 
```

But actually! `object | _T` just simplifies to `object`, since whatever `_T` is solved as being, it will always be an instance of `object` (everything in Python is). So I think we can just do this:

```suggestion
def recv(cid: SupportsIndex, default: object = ...) -> object: ...
```

Reply:
Good catch! Fixed: https://github.com/python/typeshed/pull/12199/commits/3296039dad39a977d900e4c97d7a85c338ae6065